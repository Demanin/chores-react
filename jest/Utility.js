import crypto from 'crypto';

const getRandomString = (length = 20) => {
  return crypto.randomBytes(length).toString('hex');
};

const getRandomNumber = (min = 0, max = 9999999) => {
  min = Math.ceil(min);
  max = Math.floor(max);

  return Math.floor(Math.random() * (max - min)) + min;
};

const getRandomBool = () => {
  return !!getRandomNumber(0, 1);
};

export default {
  getRandomString,
  getRandomNumber,
  getRandomBool,
};
