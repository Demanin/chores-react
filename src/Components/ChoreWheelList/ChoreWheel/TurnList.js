import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Turn from './Turn';

class TurnList extends Component
{
  render() {
    if (this.props.turnList.length === 0) {
      return null;
    }

    return (
      <div>
        {this.props.turnList.map((turn, index) =>
          <Turn
            key={index}
            turn={turn}
            onClick={() => this.props.onClick(this.props.choreWheelId, index)}
          />
        )}
      </div>
    );
  }
}

TurnList.propTypes = {
  turnList: PropTypes.arrayOf(PropTypes.string),
  onClick: PropTypes.func,
  choreWheelId: PropTypes.number,
};

export default TurnList;
