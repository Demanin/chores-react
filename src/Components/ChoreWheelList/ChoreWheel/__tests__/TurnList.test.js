import React from 'react';
import TurnList from 'src/Components/ChoreWheelList/ChoreWheel/TurnList';
import Utility from 'jest/Utility';
import { shallow } from 'enzyme';

describe('TurnList', () => {
  let onCompleteClick;
  let onSkipClick;
  let onRemoveClick;

  beforeEach(() => {
    onCompleteClick = jest.fn();
    onSkipClick = jest.fn();
    onRemoveClick = jest.fn();
  });

  test('empty wheel snapshot', () => {
    const fixture = shallow(<TurnList
      turnList={[]}
      {...{ onCompleteClick, onSkipClick, onRemoveClick }}
    />);

    expect(fixture).toMatchSnapshot();
  });

  test('single turn snapshot', () => {
    const turn = 'Random User Name';

    const fixture = shallow(<TurnList
      turnList={[turn]}
      {...{ onCompleteClick, onSkipClick, onRemoveClick }}
    />);

    expect(fixture).toMatchSnapshot();
  });

  test('single turn title', () => {
    const turn = Utility.getRandomString();

    const fixture = shallow(<TurnList
      turnList={[turn]}
      {...{ onCompleteClick, onSkipClick, onRemoveClick }}
    />);

    expect(fixture.find('h4').text()).toEqual(turn);
  });

  test('single turn complete click', () => {
    const turn = Utility.getRandomString();

    const fixture = shallow(<TurnList
      turnList={[turn]}
      {...{ onCompleteClick, onSkipClick, onRemoveClick }}
    />);

    fixture.find('ButtonGroup').childAt(0).simulate('click');

    expect(onCompleteClick).toHaveBeenCalled();
    expect(onSkipClick).not.toHaveBeenCalled();
    expect(onRemoveClick).not.toHaveBeenCalled();
  });

  test('single turn skip click', () => {
    const turn = Utility.getRandomString();

    const fixture = shallow(<TurnList
      turnList={[turn]}
      {...{ onCompleteClick, onSkipClick, onRemoveClick }}
    />);

    fixture.find('ButtonGroup').childAt(1).simulate('click');

    expect(onCompleteClick).not.toHaveBeenCalled();
    expect(onSkipClick).toHaveBeenCalled();
    expect(onRemoveClick).not.toHaveBeenCalled();
  });

  test('single turn remove click', () => {
    const turn = Utility.getRandomString();

    const fixture = shallow(<TurnList
      turnList={[turn]}
      {...{ onCompleteClick, onSkipClick, onRemoveClick }}
    />);

    fixture.find('ButtonGroup').childAt(2).simulate('click');

    expect(onCompleteClick).not.toHaveBeenCalled();
    expect(onSkipClick).not.toHaveBeenCalled();
    expect(onRemoveClick).toHaveBeenCalled();
  });

  test('two turns renders second turn below', () => {
    const turnA = Utility.getRandomString();
    const turnB = Utility.getRandomString();

    const fixture = shallow(<TurnList
      turnList={[turnA, turnB]}
      {...{ onCompleteClick, onSkipClick, onRemoveClick }}
    />);

    const expected = `<div class="text-info well-sm">${turnB}</div>`;
    expect(fixture.childAt(1).html()).toEqual(expected);
  });

  test('third and fourth turns render as tertiary turns', () => {
    const turnA = Utility.getRandomString();
    const turnB = Utility.getRandomString();
    const turnC = Utility.getRandomString();
    const turnD = Utility.getRandomString();

    const fixture = shallow(<TurnList
      turnList={[turnA, turnB, turnC, turnD]}
      {...{ onCompleteClick, onSkipClick, onRemoveClick }}
    />);

    const expectedC = `<div class="well-sm" style="color:#aaa">${turnC}</div>`;
    expect(fixture.childAt(2).html()).toEqual(expectedC);
    const expectedD = `<div class="well-sm" style="color:#aaa">${turnD}</div>`;
    expect(fixture.childAt(3).html()).toEqual(expectedD);
  });
});
