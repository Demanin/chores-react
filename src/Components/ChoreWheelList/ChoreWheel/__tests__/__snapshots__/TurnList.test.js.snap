// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TurnList empty wheel snapshot 1`] = `ShallowWrapper {}`;

exports[`TurnList single turn snapshot 1`] = `ShallowWrapper {}`;
